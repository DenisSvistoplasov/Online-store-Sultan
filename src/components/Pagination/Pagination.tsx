import React, { useMemo, useState } from 'react';
import styles from './pagination.sass';
import { classnames } from '../../utils/classnames';
import { IconArrowTickYellow } from '../icons/arrow-tick_left_yellow';

interface IPaginationProps {
  currentPage: number;
  numberOfPages: number;
  onChange: (page: number) => void;
  className?: string;
}

export function Pagination(props: IPaginationProps) {
  const {
    currentPage,
    numberOfPages,
    onChange,
    className
  } = props;

  const pages = useMemo(() => new Array(numberOfPages).fill(0).map((_, i) => i + 1), [numberOfPages]);

  const onClickPage = (page: number) => {
    onChange(page);
  };

  const onClickBack = () => {
    if (currentPage - 1 !== 0) {
      onChange(currentPage - 1);
    }
  };

  const onClickForward = () => {
    if (currentPage + 1 <= numberOfPages) {
      onChange(currentPage + 1);
    }
  };

  return (
    <div className={classnames(styles.wrapper, className)}>
      <button className={classnames(styles.back, 'interactive-btn')} onClick={onClickBack}><IconArrowTickYellow /></button>
      <ul className={styles.list}>
        {pages.map(page => (
          <li key={page} className={styles.item}>
            <button
              className={classnames(styles.page, { [styles.active]: page == currentPage }, 'interactive-btn')}
              onClick={() => onClickPage(page)}
            >{page}</button>
          </li>
        ))}
      </ul>
      <button className={classnames(styles.forward, 'interactive-btn')} onClick={onClickForward}><IconArrowTickYellow /></button>
    </div>
  );
}
