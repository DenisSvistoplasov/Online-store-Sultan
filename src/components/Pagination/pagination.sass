.wrapper
  display: flex
  justify-content: center
  align-items: center
  gap: 22px
  margin-left: auto
  margin-right: auto

.list
  display: flex
.item
.page
  display: flex
  justify-content: center
  align-items: center
  min-width: 35px
  aspect-ratio: 1/1
  border-radius: 100px
  font-size: 14px
  font-weight: 500
  &.active
    background-color: #fec
    cursor: auto
.back, .forward
  display: flex
  justify-content: center
  align-items: center
  width: 35px
  height: 35px
  border-radius: 100px
.forward
  svg
    transform: scaleX(-1)

@media ( min-width: 1140px )
  .back:hover, .forward:hover
    path
      fill: #fff
  .page:hover:not(.active)
    color: #fff