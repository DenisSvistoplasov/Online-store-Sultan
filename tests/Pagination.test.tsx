import { fireEvent, render, screen } from "@testing-library/react";
import { Pagination } from "../src/components/Pagination";
import * as react from 'react';


describe('Pagination', () => {
  it('should render', () => {
    const component = render(<Pagination currentPage={1} numberOfPages={2} onChange={() => {}} />);
    expect(component).toMatchSnapshot();
  });
  it('should properly call callback', () => {
    const onChange = jest.fn();
    render(<Pagination currentPage={2} numberOfPages={3} onChange={onChange} />);
    fireEvent.click(screen.getByText('1'));
    expect(onChange).toBeCalledWith(1);
    fireEvent.click(screen.getByText('2'));
    expect(onChange).toBeCalledWith(2);
    fireEvent.click(screen.getByTestId('back'));
    expect(onChange).toBeCalledWith(1);
    fireEvent.click(screen.getByTestId('forward'));
    expect(onChange).toBeCalledWith(3);
  })
})