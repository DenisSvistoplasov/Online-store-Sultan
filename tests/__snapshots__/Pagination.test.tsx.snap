// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <button
          class="interactive-btn"
          data-testid="back"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 9 16"
            width="9"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 2.2L3.3 8L9 13.7L7.8 16L0 8L7.8 0L9 2.2Z"
              fill="#FFC85E"
            />
          </svg>
        </button>
        <ul>
          <li>
            <button
              class="undefined interactive-btn"
            >
              1
            </button>
          </li>
          <li>
            <button
              class="interactive-btn"
            >
              2
            </button>
          </li>
        </ul>
        <button
          class="interactive-btn"
          data-testid="forward"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 9 16"
            width="9"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 2.2L3.3 8L9 13.7L7.8 16L0 8L7.8 0L9 2.2Z"
              fill="#FFC85E"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <button
        class="interactive-btn"
        data-testid="back"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 9 16"
          width="9"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 2.2L3.3 8L9 13.7L7.8 16L0 8L7.8 0L9 2.2Z"
            fill="#FFC85E"
          />
        </svg>
      </button>
      <ul>
        <li>
          <button
            class="undefined interactive-btn"
          >
            1
          </button>
        </li>
        <li>
          <button
            class="interactive-btn"
          >
            2
          </button>
        </li>
      </ul>
      <button
        class="interactive-btn"
        data-testid="forward"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 9 16"
          width="9"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 2.2L3.3 8L9 13.7L7.8 16L0 8L7.8 0L9 2.2Z"
            fill="#FFC85E"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
